tests/test_pipeline_async.py

```diff
@@ -0,0 +1,22 @@
+import pytest
+
+from psycopg import pq
+
+pytestmark = [
+    pytest.mark.asyncio,
+    pytest.mark.libpq(">=14"),
+]
+
+
+async def test_pipeline_status(aconn):
+    assert not aconn._pipeline_mode
+    async with aconn.pipeline() as p:
+        assert await p.status() == pq.PipelineStatus.ON
+        assert aconn._pipeline_mode
+        await p.sync()
+        r = aconn.pgconn.get_result()
+        assert r.status == pq.ExecStatus.PIPELINE_SYNC
```

Comment:
The reason to make `status` a property: we would have `pipeline.status()` but `result.status` :)